src/tlog/verify/body.ts

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2023 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { UnsupportedVersionError, VerificationError } from '../../error';
+import * as sigstore from '../../types/sigstore';
+import { crypto, encoding as enc } from '../../util';
+import { EntryKind, HashedRekordKind, IntotoKind } from '../types';
+
+const TLOG_MISMATCH_ERROR_MSG = 'bundle content and tlog entry do not match';
+
+// Compare the given intoto tlog entry to the given bundle
```

Comment:
Should this say intoto as it also checks hashedrekord/mesasgetype? 
```suggestion
// Compare the given tlog entry to the given bundle
```